tests/middlewares/redirect_follower_tests.rb

```diff
@@ -30,6 +30,72 @@
   env_restore
 end
 
+Shindo.tests('Excon redirector support with too many redirects') do
+  env_init
+
+  tests("request(:method => :get, :path => '/old').body").raises(Excon::Errors::TooManyRedirects) do
+    Excon.stub(
+      { :path => '/old' },
+      {
+        :headers  => { 'Location' => 'http://127.0.0.1:9292/new' },
+        :body     => 'old',
+        :status   => 301
+      }
+    )
+
+    Excon.stub(
+      { :path => '/new' },
+      {
+        :body     => 'new',
+        :status   => 200
+      }
+    )
+
+    Excon.get(
+      'http://127.0.0.1:9292',
+      :path           => '/old',
+      :redirect_limit => 0,
+      :middlewares    => Excon.defaults[:middlewares] + [Excon::Middleware::RedirectFollower],
+      :mock           => true
+    ).body
+  end
+
+  env_restore
+end
+
+Shindo.tests('Excon redirector support with redirect loop') do
+  env_init
+
+  tests("request(:method => :get, :path => '/old')").raises(Excon::Errors::TooManyRedirects) do
+    Excon.stub(
+      { :path => '/old' },
+      {
+        :headers  => { 'Location' => 'http://127.0.0.1:9292/new' },
```

Comment:
For the redirect loop, I think the location should be: `http://127.0.0.1:9292/old` again? That way it redirects to itself and gets stuck?